.gitlab-ci-check-docker-build.yml

```diff
@@ -76,4 +78,22 @@ publish:image:
     - docker login -u $REGISTRY_MENDER_IO_USERNAME -p $REGISTRY_MENDER_IO_PASSWORD registry.mender.io
     - docker login -u $DOCKER_HUB_USERNAME -p $DOCKER_HUB_PASSWORD
     - docker push $DOCKER_REPOSITORY:$COMMIT_TAG
-    - docker push $SERVICE_IMAGE
+    - docker push $DOCKER_REPOSITORY:$CI_COMMIT_REF_SLUG
+
+# Regular job, publish into user's DOCKER_REPOSITORY
+publish:image:
+  extends: .publish_image
+
+# Extra job for Enterprise repos, publish also to Docker Hub
+publish:image:dockerhub:
+  extends: .publish_image
+  only:
+    variables:
+      - $DOCKER_REPOSITORY =~ /^registry\.mender\.io.\/*/
+  before_script:
+    # The trick here is to let SERVICE_NAME use the origianl DOCKER_REPOSITORY (i.e. registry.mender.io)
```

Comment:
the only question I have is how the `SERVICE_NAME` image is now pushed, since both  calls refer to the `DOCKER_REPOSITORY` now?

Reply:
`SERVICE_NAME` is not pushed, only the "other" tags (which in fact will be the same name). I think it will work, but I guess @tranchitella will report if I am wrong :)